### ISSUE TITLE
unused variable

### ISSUE THREAD

**Issue by da0ka** [danielsaad/GCIS#1]
gc-is.hpp
comment tell us variable 'len2' is length of the previous LMS-Substring. but not used.
What is it used for?

**Comment by danielsaad** [comment]
@da0ka  You are right. The code needs refactoring. There are several unused variables which must be excluded.
